### PR TITLE
fix: amount decimals in payment description and field value

### PR DESCRIPTION
**Pull request by CzarekDryl**
https://tw.auroracreation.com/app/tasks/15647877

### CODE REVIEW

**Review by krawczykkatarzyna — DISMISSED** [review]

**Review by TomaszChmielPagepro — APPROVED** [review]

**Review by krawczykkatarzyna — APPROVED** [review]

**Review by KornelKwiatkowski — APPROVED** [review]

File: src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts [hunk 1]
```diff
@@ -50,7 +51,7 @@ export const useGetActionData = (transactionId: string | undefined) => {
         return {
           [ACTION_TYPE_FIELD_NAME]: ACTION.SIMPLE_PAYMENT,
           amount: {
-            amount,
+            amount: moveDecimal(amount, -(token?.decimals || 0)),
```
Comment (rdig) [inline src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts:54]: This needs to use `getTokenDecimalsWithFallback`

```suggestion
            amount: moveDecimal(amount, -(getTokenDecimalsWithFallback(token.decimals))),
```

File: src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts [hunk 2]
```diff
@@ -64,14 +65,17 @@ export const useGetActionData = (transactionId: string | undefined) => {
           [ACTION_TYPE_FIELD_NAME]: ACTION.SIMPLE_PAYMENT,
           from: fromDomain?.nativeId.toString(),
           amount: {
-            amount: firstPayment.amount,
+            amount: moveDecimal(firstPayment.amount, -(token?.decimals || 0)),
```
Comment (rdig) [inline src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts:68]: same

File: src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts [hunk 3]
```diff
@@ -64,14 +65,17 @@ export const useGetActionData = (transactionId: string | undefined) => {
...
             tokenAddress: firstPayment.tokenAddress,
           },
           recipient: firstPayment.recipientAddress,
           payments: additionalPayments.map((additionalPayment) => {
             return {
               amount: {
-                amount: additionalPayment.amount,
+                amount: moveDecimal(
```
Comment (rdig) [inline src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts:75]: same

**Review by rdig — CHANGES_REQUESTED** [review]
Functionally it's all good, just please change the token fallback, then I'll approve it

**Review by arrenv — DISMISSED** [review]
The number is now accurate, thank you @CzarekDryl !

I am approving, but please address rdig's feedback.

File: src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts (hunk 1 above)
Comment (CzarekDryl) [reply src/components/v5/common/ActionSidebar/hooks/useGetActionData.ts]: @rdig Fixed in all places

**Review by arrenv — APPROVED** [review]
Thanks @CzarekDryl working well!

**Comment by arrenv** [comment]
@CzarekDryl I don't think this is not introduced here, but I am seeing errors in console in this component.

image

image

image

**Comment by arrenv** [comment]
> @CzarekDryl I don't think this is introduced in your PR, but I am seeing errors in console in this component.
> 
> image
> 
> image
> 
> image

**Review by TomaszChmielPagepro — APPROVED** [review]

File: src/components/v5/common/Fields/TextareaBase/hooks.ts [hunk 4]
```diff
@@ -1,11 +1,20 @@
...
 
 const useAutosizeTextArea = (
   value: string | number | readonly string[] | undefined,
+  externalInputRef: React.ForwardedRef<HTMLTextAreaElement>,
 ) => {
   const textAreaRef = useRef<HTMLTextAreaElement>(null);
 
+  useImperativeHandle(externalInputRef, () => {
+    if (!textAreaRef.current) {
+      throw new Error('Input ref is not available');
```
Comment (TomaszChmielPagepro) [inline src/components/v5/common/Fields/TextareaBase/hooks.ts:12]: It's not `input` 😅

**Review by TomaszChmielPagepro — APPROVED** [review]

**Review by TomaszChmielPagepro — APPROVED** [review]

**Pull request merged**